cmd/kubeadm/app/cmd/join.go

```diff
@@ -250,20 +267,25 @@ func (j *Join) Run(out io.Writer) error {
 		}
 	}
 
-	kubeconfigFile := filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.KubeletBootstrapKubeConfigFileName)
+	fmt.Fprintf(out, joinDoneMsgf)
+	return nil
+}
 
-	// Write the bootstrap kubelet config file or the TLS-Boostrapped kubelet config file down to disk
-	if err := kubeconfigutil.WriteToDisk(kubeconfigFile, cfg); err != nil {
-		return err
-	}
+func getTLSBootstrappedClient() (clientset.Interface, error) {
```

Comment:
add a comment here like `// getTLSBootstrappedClient waits for the kubelet to perform the TLS bootstrap and then creates a k8s client of /etc/kubernetes/kubelet.conf`

Reply:
done.